File: app/src/main/java/com/example/rafaelanastacioalves/moby/pulllisting/PullsListAdapter.java
package com.example.rafaelanastacioalves.moby.pulllisting;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.example.rafaelanastacioalves.moby.R;
import com.example.rafaelanastacioalves.moby.vo.Pull;
import com.example.rafaelanastacioalves.moby.listeners.RecyclerViewClickListener;

import java.util.ArrayList;
import java.util.List;

class PullsListAdapter extends RecyclerView.Adapter<PullViewHolder> {
    private final Context mContext;
    private RecyclerViewClickListener recyclerViewClickListener;
    private List<Pull> items = new ArrayList<>();
    private RecyclerViewClickListener clickListener;

    public PullsListAdapter(Context context) {
        mContext = context;
    }

    private List<Pull> getItems() {
        return this.items;
    }

    public void setItems(List<Pull> data) {
        this.items = data;
        notifyDataSetChanged();
    }

    public void setRecyclerViewClickListener(RecyclerViewClickListener aRVC) {
        this.recyclerViewClickListener = aRVC;
    }

    @Override
    public PullViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new PullViewHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.pull_viewholder, parent, false), recyclerViewClickListener);
    }

    @Override
    public void onBindViewHolder(PullViewHolder holder, int position) {
        Pull aPull = getItems().get(position);
        (holder).bind(aPull, mContext);
    }

    @Override
    public int getItemCount() {
        if (getItems() != null){
            return this.items.size();

        }else{
            return 0;
        }
    }
}

File: app/src/main/java/com/desafioandroid/getirepos/view/pullslist/PullsListAdapter.kt
package com.desafioandroid.getirepos.view.pullslist

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.desafioandroid.getirepos.data.dto.PullsResponse
import com.desafioandroid.getirepos.data.utils.PullsTranslator
import com.desafioandroid.getirepos.databinding.ActivityPullsItemBinding

class PullsListAdapter(private val listener: PullsListActivityListener): RecyclerView.Adapter<PullsListAdapter.PullsViewHolder>() {
    private var pulls = ArrayList<PullsItem>()
    interface PullsListActivityListener {
        fun pullSelected(pullLink: String)
    }

    fun setPullsItems(pullsList: List<PullsResponse>) {
        this.pulls.addAll(PullsTranslator.translatePullsToPullsItem(pullsList))
        this.notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PullsViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val binding = ActivityPullsItemBinding.inflate(layoutInflater, parent, false)
        return PullsViewHolder(binding)
    }

    override fun getItemCount(): Int {
        return this.pulls.size
    }

    override fun onBindViewHolder(holder: PullsViewHolder, position: Int) {
        holder.bind(this.pulls[position])
    }

    inner class PullsViewHolder(private val binding: ActivityPullsItemBinding): RecyclerView.ViewHolder(binding.root) {
        fun bind(pull: PullsItem) {
            this.binding.pullTitleText.text = pull.title
            this.binding.userNamePullsText.text = pull.user.login
            this.binding.pullDescriptionText.text = pull.body
            Glide.with(binding.root).load(pull.user.avatarUrl).into(this.binding.userAvatarPullsImageView)
            this.binding.root.setOnClickListener {
                listener.pullSelected(pull.htmlUrl)
            }
        }
    }
}